sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -135,4 +135,14 @@ class AnalysisSuite extends AnalysisTest {
     plan = testRelation.select(CreateStructUnsafe(Seq(a, (a + 1).as("a+1"))).as("col"))
     checkAnalysis(plan, plan)
   }
+
+  test("SPARK-10534: resolve attribute references in order by clause") {
+    val a = testRelation2.output(0)
+    val c = testRelation2.output(2)
+
+    val plan = testRelation2.select(c).orderBy(Floor(a).asc)
```

Comment:
use `'c` and `'a` instead of `c` and `a`. The `plan` will get analyzed in the method `checkAnalysis`, so we should give unresolved attributes, to simulate normal query plans. (FYI, in spark SQL DSL, the symbol `'a` will be automatically turned into `UnresolvedAttribute`)